cli/deployment/config.go

```diff
@@ -557,15 +558,37 @@ func setConfig(prefix string, vip *viper.Viper, target interface{}) {
 			// with a comma, but Viper only supports with a space. This
 			// is a small hack around it!
 			rawSlice := reflect.ValueOf(vip.GetStringSlice(prefix)).Interface()
-			slice, ok := rawSlice.([]string)
+			stringSlice, ok := rawSlice.([]string)
 			if !ok {
 				panic(fmt.Sprintf("string slice is of type %T", rawSlice))
 			}
-			value := make([]string, 0, len(slice))
-			for _, entry := range slice {
+			value := make([]string, 0, len(stringSlice))
+			for _, entry := range stringSlice {
 				value = append(value, strings.Split(entry, ",")...)
 			}
 			val.FieldByName("Value").Set(reflect.ValueOf(value))
+		case codersdk.Experiments:
+			// As []string above, but we support setting wildcard values
+			// '*' or 'true' to enable experiments listed in codersdk.ExperimentsAll.
+			// Experiments not listed in codersdk.ExperimentsAll must be enabled
+			// explicitly.
+			vip.MustBindEnv(prefix, env)
+			rawSlice := reflect.ValueOf(vip.GetStringSlice(prefix)).Interface()
+			stringSlice, ok := rawSlice.([]string)
+			if !ok {
+				panic(fmt.Sprintf("string slice is of type %T", rawSlice))
+			}
+			value := make([]codersdk.Experiment, 0, len(stringSlice))
+			for _, entry := range stringSlice {
+				for _, val := range strings.Split(entry, ",") {
+					if val == "*" || val == "true" {
+						value = append(value, codersdk.ExperimentsAll...)
+					} else {
+						value = append(value, codersdk.Experiment(val))
+					}
+				}
+			}
```

Comment:
I think the value should be a `[]string` in deployment config without any extra munging/parsing. The deployment config package in my mind just does reading and unmarshaling of values from env/flags and doesn't do any validation or mutation.

Perhaps you could add a package like `experiments` with `ParseFlags(raw []string) []codersdk.Experiment` and `GetFlag(raw []string, flag codersdk.Experiment) bool`. These operations aren't very expensive so I wouldn't even bother with caching the values TBH.

Reply:
I'm not super fixated on this suggestion, but we don't do processing for any other values in this package. I think that if we added processing for all of our flags here this package will get very large.

Reply:
This is done.